### PR TITLE
Fix BoolValueForPB may be nil

### LINKED ISSUE #31888: Not providing "enabled" for ingressGateway in istoctl yaml causes panic: runtime error: invalid memory address or nil pointer dereference
**Bug description**
When omitting "enabled" key for ingressGateways causes istioctl to crash with following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x28252d7]

goroutine 1 [running]:
istio.io/istio/operator/cmd/mesh.getProfileNSAndEnabledComponents(0xc000b40000, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0xc0005dd440, 0x35e55c0, 0xc000f76480)
        istio.io/istio/operator/cmd/mesh/install.go:327 +0x97
istio.io/istio/operator/cmd/mesh.runApplyCmd(0xc000f162c0, 0xc000ec115d, 0xc000d47600, 0xc00068a990, 0x0, 0x0)
        istio.io/istio/operator/cmd/mesh/install.go:155 +0x365
istio.io/istio/operator/cmd/mesh.InstallCmd.func2(0xc000f162c0, 0xc000efaba0, 0x0, 0x2, 0x0, 0x0)
        istio.io/istio/operator/cmd/mesh/install.go:122 +0x46
github.com/spf13/cobra.(*Command).execute(0xc000f162c0, 0xc000efab80, 0x2, 0x2, 0xc000f162c0, 0xc000efab80)
        github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000ec8580, 0x3, 0x3, 0xc000ec8580)
        github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.1.1/command.go:895
main.main()
        istio.io/istio/istioctl/cmd/istioctl/main.go:36 +0x128
```

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Not including "enabled" key either produces error messages indicating it is required, or appropriate default is used (docs suggest key is not erquired)

**Steps to reproduce the bug**
Here is a minimal yaml file to use with `istioctl install -f <this yaml>.yaml`
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    ingressGateways:
      - name: istio-egressgateway
```

Adding "enabled" key avoids the crash:
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    ingressGateways:
      - name: istio-egressgateway
        enabled: true
```

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

istioctl
```
client version: 1.9.2
control plane version: 1.9.2
data plane version: 1.8.0 (49 proxies), 1.9.2 (4 proxies)
```
kubectl
```
Client Version: v1.18.0
Server Version: v1.17.17-gke.1101
```

**How was Istio installed?**
Using istioctl

**Environment where the bug was observed (cloud vendor, OS, etc)**
GKE 1.17.17-gke.1101

local environment: Ubuntu 20.04 on WSL2 on windows 10

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

### PR DESCRIPTION
**Pull request by kebe7jun**
For: https://github.com/istio/istio/issues/31888

[X] Installation

### CODE REVIEW

File: operator/pkg/util/util.go [hunk 1]
```diff
@@ -140,3 +142,11 @@ func RenderTemplate(tmpl string, ts interface{}) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+// BoolValue returns true is v is not nil and v.Value is true, or false otherwise.
```
Comment (shamsher31) [inline operator/pkg/util/util.go:146]:
```suggestion
// BoolValue returns true if v is not nil and v.Value is true, or false otherwise.
```

**Review by shamsher31 — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

File: operator/pkg/util/util.go [hunk 2]
```diff
@@ -140,3 +142,11 @@ func RenderTemplate(tmpl string, ts interface{}) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+// BoolValue returns true if v is not nil and v.Value is true, or false otherwise.
+func BoolValue(v *v1alpha1.BoolValueForPB) bool {
```
Comment (howardjohn) [inline operator/pkg/util/util.go:147]: Can we remove this? You can just do v.GetValue() which is more clear and already nil-safe

Comment (kebe7jun) [reply operator/pkg/util/util.go:147]: `v.GetValue()` may be not work, due to:
https://github.com/istio/api/blob/a9c5294cac41c9089557ef1464f8353c5ef2b346/operator/v1alpha1/common.go#L60-L62

If we choose to go that way, we may need to modify there.

Comment (kebe7jun) [reply operator/pkg/util/util.go:147]: Added a new pr for this https://github.com/istio/api/issues/1938

**Pull request merged**